### PR TITLE
[docs] Add missing pickers migration docs

### PR DESCRIPTION
**Pull request by LukasTy**
Add missing breaking changes from #6543 and #6445.

[Docs preview](https://deploy-preview-7000--material-ui-x.netlify.app/x/migration/migration-pickers-v5/#replace-toolbar-props-by-slot)

P.S. I've already updated the GH pre-release with missing info (about tabs)

### CODE REVIEW

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 1]
```diff
@@ -304,3 +304,93 @@ Component name changes are also reflected in `themeAugmentation`:
...
+  +    tabs: {
+  +      hidden: false,
+  +      dateRangeIcon: <LightModeIcon />,
+  +      timeIcon: <AcUnitIcon />,
+  +    }
+  +  }}
+   />
+  ```
+
+- The `onChange` prop on `DateTimePickerTabs` component has been renamed to `onViewChange` to better fit its usage:
```
Comment (flaviendelangle) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:389]: The actual breaking change would be if user try to set a custom tab component
I would rephrase it the opposite way

Something like

````suggestion
- The `onChange` prop of the `tabs` component slot has been renamed `onViewChange`

```diff
const CustomTabComponent = props => (
  <div>
-  <button onClick={() => props.onChange('day')}>Show day view</button>
+  <button onClick={() => props.onViewChange('day')}>Show day view</button>
  </div>
)

<DateTimePicker
  components={{
    tabs: CustomTabComponent
  }}
/>
```
````

Comment (LukasTy) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:389]: Or they have used our publicly exported `DateTimePickerTabs` to build something custom. 🤔 
I'd say that such changes we could include both the current diff and my suggestion—possibly leaving the text the same..? 🤔

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 2]
```diff
@@ -304,3 +304,93 @@ Component name changes are also reflected in `themeAugmentation`:
...
+  ```diff
+   // Same on all other Date Time picker variations
+   <DateTimePicker
+  -  hideTabs={false}
+  -  dateRangeIcon={<LightModeIcon />}
+  -  timeIcon={<AcUnitIcon />}
+  +  componentsProps={{
+  +    tabs: {
+  +      hidden: false,
+  +      dateRangeIcon: <LightModeIcon />,
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:382]: This icon name is a bit strange to me.
What do you think about renaming it to `dateIcon` to be more in line with `timeIcon` @alexfauquette @flaviendelangle ? 🤔

Comment (flaviendelangle) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:382]: The name of the prop come from the name of the default icon
But I do agree that in the context of the picker it's very confusing

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md (hunk 1 above)
Comment (flaviendelangle) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:389]: You are right, we could do both

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md (hunk 2 above)
Comment (alexfauquette) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:382]: +1 for changing to `dateIcon` which would feel more natural. Better having a meaning full prop naming than to match the default icon name

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md (hunk 1 above)
Comment (LukasTy) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:389]: Updated 😉

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md (hunk 2 above)
Comment (LukasTy) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:382]: Thank you for the comments.
I've [created a PR](https://github.com/mui/mui-x/pull/7024) to handle it.
Just waiting for this one to be merged firstly. 👌

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 3]
```diff
@@ -19,6 +19,24 @@ Since v6 is a major release, it contains some changes that affect the public API
 These changes were done for consistency, improve stability and make room for new features.
 Below are described the steps you need to make to migrate from v5 to v6.
 
+### Drop `clock` in desktop mode
+
+We have dropped the clock picker behavior in desktop mode for `DateTimePicker` and `TimePicker` components.
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:24]: @joserodolfofreitas or any1 else—feel free to suggest improvements on this or the below sections. 🙏

Comment (alexfauquette) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:24]: In desktop mode, the `DateTimePicker` and `TimePicker` components will not display the clock.

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 4]
```diff
@@ -19,6 +19,24 @@ Since v6 is a major release, it contains some changes that affect the public API
...
+This is the first step towards moving to a [better implementation](https://github.com/mui/mui-x/issues/4483).
+The behavior on mobile mode is still the same.
+If you were relying on Clock Picker in desktop mode for tests—make sure to check [testing caveats](/x/react-date-pickers/getting-started/#testing-caveats) to choose the best replacement for it.
+
+### Drop direct `@date-io` adapters support
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:29]: It sounds like we are dropping `@date-io`

```suggestion
### Extended `@date-io` adapters
```

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 5]
```diff
@@ -19,6 +19,24 @@ Since v6 is a major release, it contains some changes that affect the public API
...
+
+In order to support new Date and Time pickers with [fields components](/x/react-date-pickers/fields/) we had to augment `@date-io` adapters.
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:31]: In v5, it was possible to import adapters either from `@date-io` or `@mui/x-date-pickers` which were the same.
In v6, the adapters are extended by `@mui/x-date-pickers` to support field components. 
Which means adapters can not be imported from `@date-io` anymore. They might be imported from `@mui/x-date-pickers` or `@mui/x-date-pickers-pro`. Otherwise, some methods will be missing.

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 6]
```diff
@@ -19,6 +19,24 @@ Since v6 is a major release, it contains some changes that affect the public API
...
+If you were using adapter from `@date-io`—you will need to update the import.
+If we do not have a specific augmented adapter you were using—there probably was a reason for it, but you can raise an issue expressing interest in it.
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:33]:
```suggestion
If you do not find the adapter you were using—there probably was a reason for it, but you can raise an issue expressing interest in it.
```

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md (hunk 3 above)
Comment (LukasTy) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md]: Updated 😉

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md (hunk 5 above)
Comment (LukasTy) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md]: Updated with some tweaks to the whole paragraph. 👌

**Review by flaviendelangle — APPROVED** [review]
Nice addition :+1:
